library/src/main/java/com/novoda/downloadmanager/UnlinkedDataRemover.java

```diff
@@ -0,0 +1,20 @@
+package com.novoda.downloadmanager;
+
+class UnlinkedDataRemover {
+    private final LocalFilesDirectory localFilesDirectory;
+    private final V2DatabaseFiles v2DatabaseFiles;
+
+    UnlinkedDataRemover(LocalFilesDirectory localFilesDirectory, V2DatabaseFiles v2DatabaseFiles) {
+        this.localFilesDirectory = localFilesDirectory;
+        this.v2DatabaseFiles = v2DatabaseFiles;
+    }
+
+    void remove() {
+        for (String filename : localFilesDirectory.contents()) {
+            if (!v2DatabaseFiles.databaseContents().contains(filename)) {
```

Comment:
This method kind of represents my point from before. It doesn't need to know or care that the backing data-structure is a database.

`if(!downloadFiles.fileNames().contains(fileName)))`